server.js

```diff
@@ -1,32 +1,65 @@
 import express from "express";
 import cors from "cors";
 import mongoose from "mongoose";
+import listEndpoints from "express-list-endpoints";
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
+import booksData from "./data/books.json";
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
 mongoose.Promise = Promise;
 
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
 const port = process.env.PORT || 8080;
 const app = express();
 
-// Add middlewares to enable cors and json body parsing
+const Book = mongoose.model("Book", {
+  bookID: Number,
+  title: String,
+  authors: String,
+  average_rating: Number,
+  isbn: Number,
+  isbn13: Number,
+  language_code: String,
+  num_pages: Number,
+  ratings_count: Number,
+  text_reviews_count: Number,
+});
+
+if (process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    await Book.deleteMany();
+    booksData.forEach((singleBook) => {
+      const newBook = new Book(singleBook);
+      newBook.save();
+    });
+  };
+  seedDatabase();
+}
+
 app.use(cors());
 app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send(listEndpoints(app));
+});
+
+//1. endpoint that returns the whole array of book ratings.
+app.get("/bookratings", async (req, res) => {
+  const books = await Book.find(req.query);
+  res.send(books);
+});
+
+// 2. endpoint that returns one rated book by its title.
+app.get("/bookratings/booktitle/:title", async (req, res) => {
+  const bookByTitle = await Book.findOne({ title: req.params.title });
+  res.send(bookByTitle);
+});
+
+// // 3. endpoint that returns all rated books written by a specific author.
+app.get("/bookratings/author/:authors", async (req, res) => {
+  const bookByAuthor = await Book.find({ authors: req.params.authors });
+  res.send(bookByAuthor);
```

Comment:
Great job returning an array of (multiple) books by one author.
The only suggestion I have would be to allocate for case sensitivity.
In this case, the easiest would be to use` RegExp`

For example: 
Line 61, we could do something like this:

`const bookByAuthor = await Book.find({ 
authors: new RegExp (req.params.authors, 'i')
});`

Or save req.params.authors in a variable and replace it with the variable like:

```
const findByAuthors = req.params.authors
const bookByAuthor = await Book.find({ 
authors: new RegExp (findByAuthors, 'i')
});
```
We can also destructure the req.params instead of a new variable

```
const { authors } = req.params
const bookByAuthor = await Book.find({ 
authors: new RegExp (authors, 'i')
});
```